sympy/core/exprtools.py

```diff
@@ -355,6 +354,12 @@ def __init__(self, factors=None):  # Factors
             for _ in range(i):
                 c.remove(I)
             factors = dict(Mul._from_args(c).as_powers_dict())
+            # Handle all rational Coefficients
+            for f in list(factors.keys()):
```

Comment:
Would it suffice to write `for f in factors.keys():`?

Reply:
No, the `factors.keys()` method returns an iterator and as I change the dictionary the iterator throws an error.